monitor/basic/config_test.go

```diff
@@ -0,0 +1,52 @@
+package basic
+
+import (
+	"encoding/json"
+	"testing"
+)
+
+var cfgJSON = []byte(`
+{
+      "check_interval": "15s"
+}
+`)
+
+func TestLoadJSON(t *testing.T) {
+	cfg := &Config{}
+	err := cfg.LoadJSON(cfgJSON)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	j := &jsonConfig{}
+
+	json.Unmarshal(cfgJSON, j)
+	j.CheckInterval = "-10"
+	tst, _ := json.Marshal(j)
+	err = cfg.LoadJSON(tst)
+	if err == nil {
+		t.Error("expected error decoding check_interval")
+	}
+}
+
+func TestToJSON(t *testing.T) {
+	cfg := &Config{}
+	cfg.LoadJSON(cfgJSON)
+	_, err := cfg.ToJSON()
```

Comment:
Cannot match directly against cfgJSON because indentation or key order might be changed, but it's probably a good thing to re-parse the generated json and check it validates.